MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,182 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionOpacity;
+
+/**
+ * Change the bearing and tilt of camera when viewing extrusions to get full 3D effect
+ */
+public class RotationExtrusionActivity extends AppCompatActivity implements SensorEventListener {
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private SensorManager sensorManager;
+  private Sensor gyro;
+  private Sensor magnetic;
+  private float[] gravArray;
+  private float[] magneticArray;
+  private float azimut;
+  private float pitch;
+  private float roll;
+  private float[] inclinationMatrix;
+  private float[] rotationMatrix;
+
+  //amplifiers that translate small movements in phone orientation into larger viewable map changes
+  //pitch is negative to compensate for the negative readings from the device while face up
+  //90 is used based on the viewable angle when viewing the map (from phone being flat to facing you
+  private int pitchAmplifier = -90;
+  private int bearingAmplifier = 90;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_extrusion_rotation);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+        @Override
+        public void onMapReady(@NonNull final MapboxMap map) {
+          mapboxMap = map;
+          setupBuildings();
+        }
+    });
+
+    //initialize sensors
+    sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+    gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+    magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+
+  }
+
+  private void setupBuildings() {
+    FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+    fillExtrusionLayer.setSourceLayer("building");
+    fillExtrusionLayer.setFilter(eq("extrude", "true"));
+    fillExtrusionLayer.setMinZoom(15);
+    fillExtrusionLayer.setProperties(
+            fillExtrusionColor(Color.LTGRAY),
+            fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+            fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+            fillExtrusionOpacity(0.9f)
+    );
+    mapboxMap.addLayer(fillExtrusionLayer);
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    sensorManager.registerListener(this, gyro , 200);
+    sensorManager.registerListener(this, magnetic , 200);
```

Comment:
In order to increase readability, what about replacing magic numbers with constants with a particular meaning?
Further information: [Replace Magic Number with Symbolic Constant](https://refactoring.com/catalog/replaceMagicNumberWithSymbolicConstant.html).